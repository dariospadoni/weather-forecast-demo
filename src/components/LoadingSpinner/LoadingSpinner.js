import React from 'react';
import './LoadingSpinner.scss';

const LoadingSpinner = () => 
  <div className="spinner">
    <label className="spinner__label">
      Loadingâ€¦ 
    </label>
    <progress></progress>
  </div>;
  
export default LoadingSpinner;
