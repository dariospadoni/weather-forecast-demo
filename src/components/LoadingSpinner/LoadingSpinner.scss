.spinner {
  position: fixed; /* or absolute */
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
  text-align: center;
  font-size: 36px;

  & .spinner__label {
    padding: 12px;
    display: block;
    font-weight: 700;
  }
}
