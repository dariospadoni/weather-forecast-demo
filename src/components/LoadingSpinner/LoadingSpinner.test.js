import React from 'react';
import { render } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';
 
it('renders correctly', () => {
  const { container, getByText } = render(<LoadingSpinner />);
  const loadingText = getByText(/Loadingâ€¦/);
  expect(loadingText).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
