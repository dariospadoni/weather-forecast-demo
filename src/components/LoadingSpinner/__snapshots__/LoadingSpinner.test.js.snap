// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="spinner"
  >
    <label
      class="spinner__label"
    >
      Loadingâ€¦
    </label>
    <progress />
  </div>
</div>
`;
